operators/src/source/ogr_source.rs

```diff
@@ -0,0 +1,2856 @@
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+use std::fs::File;
+use std::iter::Peekable;
+use std::marker::PhantomData;
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::str::FromStr;
+use std::sync::mpsc::{self, Receiver, SyncSender, TryRecvError, TrySendError};
+use std::sync::Arc;
+use std::task::Poll;
+
+use chrono::DateTime;
+use futures::stream::BoxStream;
+use futures::task::{Context, Waker};
+use futures::Stream;
+use futures::StreamExt;
+use gdal::vector::{Dataset, Feature, FeatureIterator, FieldValue, OGRwkbGeometryType};
+use serde::{Deserialize, Serialize};
+use tokio::task::spawn_blocking;
+
+use geoengine_datatypes::collections::{
+    BuilderProvider, FeatureCollection, FeatureCollectionBuilder, FeatureCollectionRowBuilder,
+    GeoFeatureCollectionRowBuilder, VectorDataType,
+};
+use geoengine_datatypes::primitives::{
+    Coordinate2D, FeatureDataType, FeatureDataValue, Geometry, MultiLineString, MultiPoint,
+    MultiPolygon, NoGeometry, TimeInstance, TimeInterval, TypedGeometry,
+};
+use geoengine_datatypes::provenance::ProvenanceInformation;
+use geoengine_datatypes::spatial_reference::SpatialReference;
+use geoengine_datatypes::util::arrow::ArrowTyped;
+
+use crate::engine::{
+    InitializedOperator, InitializedOperatorImpl, QueryContext, QueryProcessor, QueryRectangle,
+    SourceOperator, TypedVectorQueryProcessor, VectorOperator, VectorQueryProcessor,
+    VectorResultDescriptor,
+};
+use crate::error::Error;
+use crate::util::Result;
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct OgrSourceParameters {
+    pub layer_name: String,
+    pub attribute_projection: Option<Vec<String>>,
+}
+
+pub type OgrSource = SourceOperator<OgrSourceParameters>;
+
+///  - `filename`: path to the input file
+///  - `layer_name`: name of the layer to load
+///  - `time`: the type of the time attribute(s)
+///  - `duration`: the duration of the time validity for all features in the file [if time == "duration"]
```

Comment:
I copied it 1:1 from MAPPING. We could decide to change it, though.

Reply:
I think we should

Reply:
Ok, I never had the feeling that our goal was to improve all operator configs but to re-use as much as possible. What is our gouidline?

Reply:
I never knew we had the intention of being API compatible to Mapping ;) Otherwise we would need to return 200 OK always ;) 

I think the guideline should be to provide clear, simple, extensible, failure-proof, correct APIs. And having variables depending on other variables where user input errors are only caught deep inside application logic doesn't fit these criteria.

Reply:
9c8d595

Reply:
please update documentation accordingly

Reply:
93b7dcb